elrond-wasm/src/api/send_api.rs

```diff
@@ -115,90 +138,104 @@ pub trait SendApi: Clone + Sized {
     /// Use a `HexCallDataSerializer` to prepare this field.
     fn async_call_raw(
         &self,
-        to: &Address,
+        to: &ManagedAddress<Self::ProxyTypeManager>,
         amount: &BigUint<Self::ProxyTypeManager>,
-        data: &[u8],
+        endpoint_name: &ManagedBuffer<Self::ProxyTypeManager>,
+        arg_buffer: &ManagedArgBuffer<Self::ProxyTypeManager>,
     ) -> !;
 
-    /// Sends an asynchronous call to another contract, with either EGLD or ESDT value.
-    /// The `token` argument decides which one it will be.
-    /// Calling this method immediately terminates tx execution.
-    fn async_call(&self, async_call: AsyncCall<Self>) -> ! {
-        self.async_call_raw(
-            &async_call.to,
-            &async_call.egld_payment,
-            async_call.hex_data.as_slice(),
-        )
-    }
+    // /// Sends an asynchronous call to another contract, with either EGLD or ESDT value.
+    // /// The `token` argument decides which one it will be.
+    // /// Calling this method immediately terminates tx execution.
+    // fn async_call(&self, async_call: AsyncCall<Self>) -> ! {
+    //     self.async_call_raw(
+    //         &async_call.to,
+    //         &async_call.egld_payment,
+    //         async_call.hex_data.as_slice(),
+    //     )
+    // }
 
     /// Performs a simple ESDT/NFT transfer, but via async call.  
     /// As with any async call, this immediately terminates the execution of the current call.  
     /// So only use as the last call in your endpoint.  
     /// If you want to perform multiple transfers, use `self.send().transfer_multiple_esdt_via_async_call()` instead.  
     /// Note that EGLD can NOT be transfered with this function.  
-    fn transfer_esdt_via_async_call(
+    fn transfer_esdt_via_async_call<D>(
         &self,
-        to: &Address,
+        to: &ManagedAddress<Self::ProxyTypeManager>,
         token: &TokenIdentifier<Self::ProxyTypeManager>,
         nonce: u64,
         amount: &BigUint<Self::ProxyTypeManager>,
-        data: &[u8],
-    ) -> ! {
+        data: D,
+    ) -> !
+    where
+        D: ManagedInto<Self::ProxyTypeManager, ManagedBuffer<Self::ProxyTypeManager>>,
+    {
+        let data_buf: ManagedBuffer<Self::ProxyTypeManager> =
+            data.managed_into(self.type_manager());
         if nonce == 0 {
-            let mut serializer = HexCallDataSerializer::new(ESDT_TRANSFER_STRING);
-            serializer.push_argument_bytes(token.to_esdt_identifier().as_slice());
-            serializer.push_argument_bytes(amount.to_bytes_be().as_slice());
-            if !data.is_empty() {
-                serializer.push_argument_bytes(data);
+            let mut arg_buffer = ManagedArgBuffer::new_empty(self.type_manager());
```

Comment:
Clippy complains "all if blocks contain the same code at the start ". 

Move the `arg_buffer` declaration outside the `if` statement?